lib/fastly_nsq/message_queue/consumer.rb

```diff
@@ -10,18 +10,21 @@ class Consumer
     def_delegator :connection, :size
     def_delegator :connection, :terminate
 
-    def initialize(topic:, channel:, ssl_context: nil)
-      @topic = topic
-      @channel = channel
+    def initialize(topic:, channel:, ssl_context: nil, &connector)
+      @topic       = topic
+      @channel     = channel
       @ssl_context = SSLContext.new(ssl_context)
+      @connector   = connector || DEFAULT_CONNECTOR
```

Comment:
Just my two cents, I kind of liked this PR before this change.

This feels now probably more general than we need it to be, although it seems to make testing easier.

Also, and this is not scientific or a well-described perspective by me (sorry), but I haven't seen the constructor take a block and it seems a little funky. I'd expect either the connector to be passed in as an arg or have some other method take a block.

So I'm a -1 on this recent change, but it's like a 3 out of 10 for me.